services/drupal/web/modules/custom/epa_migrations/src/EpaMediaWysiwygTransformTrait.php

```diff
@@ -77,14 +67,27 @@ public function transformWysiwyg($wysiwyg_content, EntityTypeManager $entityType
           );
         }
         else {
+          $doc = new DOMDocument();
+          $el = $doc->createElement('drupal-media');
+          $el->setAttribute('data-entity-type', 'media');
+          $el->setAttribute('data-entity-uuid', $media_entity_uuid);
+          $el->setAttribute('data-view-mode', $view_modes[$tag_info['view_mode']]);
+
           $alignment = $remove_alignment ? '' : $tag_info['fields']['field_image_alignment[und]'] ?? 'center';
-          return sprintf($media_embed_replacement_template,
-            $tag_info['fields']['field_file_image_alt_text[und][0][value]'] ?? '',
-            $alignment,
-            htmlentities(stripslashes(urldecode($tag_info['fields']['field_caption[und][0][value]']))) ?? '',
-            $media_entity_uuid,
-            $view_modes[$tag_info['view_mode']]
-          );
+          $el->setAttribute('data-align',$alignment);
+
+          $caption = htmlentities(stripslashes(urldecode($tag_info['fields']['field_caption[und][0][value]']))) ?? '';
+          if (!empty($caption)) {
+            $el->setAttribute('data-caption',$caption);
+          }
```

Comment:
```suggestion
          $caption = stripslashes(urldecode($tag_info['fields']['field_caption[und][0][value]'] ?? ''));
          if (!empty($caption)) {
            $el->setAttribute('data-caption', $caption);
          }
```

This does two things: moves `??` expression so notice won't occur calling urldecode, and removes the now unnecessary htmlentities.